src/SigasiProjectCreator/ArgsAndFileParser.py

```diff
@@ -4,37 +4,40 @@
     :license: BSD, see LICENSE for more details.
 """
 import os
-from optparse import OptionParser
+import argparse
 
 
 class ArgsAndFileParser:
     options = None
 
     def __init__(self, usage):
-        self.parser = OptionParser(usage=usage)
-        self.parser.add_option('-l', '--layout', action="store", dest="layout",
-                               choices=['default', 'simulator'], default='default')
-
-    def parse_args(self, args_len, optional_dir=False):
-        (ArgsAndFileParser.options, args) = self.parser.parse_args()
-
-        if len(args) < args_len:
-            self.parser.error("incorrect number of arguments")
-        if optional_dir:
-            self.check_next_arg_is_dir(args, args_len)
+        self.parser = argparse.ArgumentParser(prog='SigasiProjectCreator')
+        self.parser.add_argument('project_name', help='project name')
+        self.parser.add_argument('input_file', help='input file')
+        self.parser.add_argument('destination_folder', help='root folder of created project')
+        self.parser.add_argument('-l', '--layout', action='store', dest='layout',
+                                 choices=['default', 'simulator'], default='default',
+                                 help='Project layout: default (in place) or simulator (one folder per library with linked files)')
+
+    def parse_args(self):
+        args = self.parser.parse_args()
+        if args.project_name is None:
```

Comment:
According to https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_argument, you can add `required=True` and you'd not have to check this here.

Reply:
Nice. Code updated